### PR TITLE
feat: Add `authorId` query param to `/blocks` endpoint

### LINKED ISSUE #288: Add query param to opt out of authorId field for `/blocks`
Currently we always provide the ID of the blocks author. This has the overhead of a relatively large RPC call for the current validator set. For users concerned with performance it would be beneficial to be able to opt out of this option.

### PR DESCRIPTION
**Pull request by emostov**
Closes #288

This PR introduces the ~`noAuthor`~ `authorId` query param to the `/blocks` endpoint. 

For some users the `authorId` field may not be necessary, and since it requires a rather slow RPC query to the validators storage item in the sessions pallet, I believe it makes sense to give the option to opt out of it.

---
Below shows the difference in response times I got when using `noAuthor=true`
```
 📦 SUBSTRATE:
     ✅ WS_URL: "wss://rpc.polkadot.io"
2020-09-18 14:17:23 info: Connected to chain Polkadot on the parity-polkadot client at wss://rpc.polkadot.io
2020-09-18 14:17:23 info: Listening on http://127.0.0.1:8080/
2020-09-18 14:17:25 http: GET /blocks/1603921 200 1408ms
2020-09-18 14:17:31 http: GET /blocks/1603921 200 2009ms
2020-09-18 14:17:35 http: GET /blocks/1603921 200 1346ms
2020-09-18 14:17:37 http: GET /blocks/1603921 200 1322ms
2020-09-18 14:17:39 http: GET /blocks/1603921 200 1326ms
2020-09-18 14:17:55 http: GET /blocks/1603921?noAuthor=true 200 1229ms
2020-09-18 14:17:59 http: GET /blocks/1603921?noAuthor=true 200 1205ms
2020-09-18 14:18:01 http: GET /blocks/1603921?noAuthor=true 200 1256ms
2020-09-18 14:18:04 http: GET /blocks/1603921?noAuthor=true 200 1207ms
2020-09-18 14:18:06 http: GET /blocks/1603921?noAuthor=true 200 1225ms
2020-09-18 14:18:07 http: GET /blocks/1603921?noAuthor=true 200 1196ms
2020-09-18 14:18:14 http: GET /blocks/1603921 200 1375ms
2020-09-18 14:18:16 http: GET /blocks/1603921 200 1673ms
2020-09-18 14:18:19 http: GET /blocks/1603921 200 1358ms
2020-09-18 14:18:21 http: GET /blocks/1603921 200 1362ms
2020-09-18 14:18:23 http: GET /blocks/1603921 200 1651ms
2020-09-18 14:18:25 http: GET /blocks/1603921 200 1337ms
2020-09-18 14:18:28 http: GET /blocks/1603921 200 1318ms
2020-09-18 14:19:02 http: GET /blocks/1603921?noAuthor=true 200 1209ms
2020-09-18 14:19:04 http: GET /blocks/1603921?noAuthor=true 200 1200ms
2020-09-18 14:19:06 http: GET /blocks/1603921?noAuthor=true 200 1195ms
2020-09-18 14:19:08 http: GET /blocks/1603921?noAuthor=true 200 1199ms

### CODE REVIEW

File: docs/src/openapi-v1.yaml [hunk 1]
```diff
@@ -242,13 +242,25 @@ paths:
...
+          default: false
       - name: extrinsicDocs
         in: query
         description: When set to `true`, every extrinsic will have an extra `docs`
           property with a string of the extrinsics documentation.
         required: false
         schema:
           type: boolean
+          default: false
+      - name: noAuthor
```
Comment (danforbes) [inline docs/src/openapi-v1.yaml:254]: I don't really feel super strongly about this, but it seems a little like a double-negative to call this `noAuthor` and then set the default to `false`. What if we called the param `authorId` and said the default value was `true`? I don't know...like I said, just a thought.

File: src/services/blocks/BlocksService.ts [hunk 2]
```diff
@@ -42,15 +42,24 @@ export class BlocksService extends AbstractService {
 	async fetchBlock(
 		hash: BlockHash,
 		eventDocs: boolean,
-		extrinsicDocs: boolean
+		extrinsicDocs: boolean,
+		noAuthor: boolean
```
Comment (danforbes) [inline src/services/blocks/BlocksService.ts:46]: Similar comment - should this be renamed `author`?

**Review by danforbes — APPROVED** [review]

File: docs/src/openapi-v1.yaml (hunk 1 above)
Comment (emostov) [reply docs/src/openapi-v1.yaml:254]: Great point, will change 👍

**Comment by emostov** [comment]
Updated so the query param is `authorId`. The param defaults to true and needs to bet to false in order to exclude the `authorId`.

**Review by joepetrowski — APPROVED** [review]

**Comment by dvdplm** [comment]
I'm late to the game here (apologies). I don't think this is a good change: it's an API change made mostly to paper over internal performance (or lack thereof). I think it's better to keep the API correct and complete and fix performance problems elsewhere.

Is this released already?

**Pull request merged**